src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,46 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+
 public class SalaryInfo {
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        StringBuilder stringBuilder = new StringBuilder(
+                "Report for period " + dateFrom + " - " + dateTo);
+        int space = 1;
+        LocalDate localDateFrom = LocalDate.of(
+                Integer.parseInt(dateFrom.substring(6, 10)),
+                Integer.parseInt(dateFrom.substring(3, 5)),
+                Integer.parseInt(dateFrom.substring(0, 2))
+        );
+        LocalDate localDateTo = LocalDate.of(
+                Integer.parseInt(dateTo.substring(6, 10)),
+                Integer.parseInt(dateTo.substring(3, 5)),
+                Integer.parseInt(dateTo.substring(0, 2))
+        );
+        for (String name : names) {
+            int salary = 0;
+            for (String oneData : data) {
+                String[] oneDataSplit = oneData.split(" ");
+                LocalDate localOneData = LocalDate.of(
+                        Integer.parseInt(oneDataSplit[0].substring(6, 10)),
+                        Integer.parseInt(oneDataSplit[0].substring(3, 5)),
+                        Integer.parseInt(oneDataSplit[0].substring(0, 2))
+                );
+                String trimmedName = oneDataSplit[1];
```

Comment:
please extract data indexes to constant ("DATE_INDEX" = 0, "NAME_INDEX" = 1, etc).